lib/Semantics/check-select-stmt.cpp

```diff
@@ -0,0 +1,398 @@
+//===-- lib/semantics/check-select-stmt.cc --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "check-select-stmt.h"
+#include "flang/Parser/message.h"
+#include "flang/Semantics/tools.h"
+
+namespace Fortran::semantics {
+
+class SelectCaseHelper {
+
+public:
+  SelectCaseHelper(parser::Messages &messages) : messages_{messages} {}
+  void SetKind(int kind) { kind_ = kind; }
+  void SetTypeCategory(TypeCategory category) { category_ = category; }
+
+  template<typename T> std::string GetValueAsString(T val) {
+    if constexpr (std::is_same_v<T, bool>) {
+      return std::string(val ? "true" : "false");
+    } else if constexpr (std::is_same_v<T, std::int64_t>) {
+      return std::string(std::to_string(val));
```

Comment:
```suggestion
      return std::to_string(val);
```

Reply:
I have incorporated this review comment.